File: ene-jun-2023/MiguelAlejandroAnguiano/Practica6.py
def main():
    n = int(input())
    participantes=[]

    for _ in range(n):
        s, x, y, z = input().split()
        x, y, z = int(x), int(y), int(z)
        participantes.append((s, x, y, z))

    quicksort(participantes, 0, n-1)

    for s, x, y, z in participantes:
        print(s, x, y, z)

def quicksort(arr, low, high):
    if low < high:
        pivot = partition(arr, low, high)
        quicksort(arr, low, pivot - 1)
        quicksort(arr, pivot + 1, high)

def partition(arr, low, high):
    pivot = arr[high][1:]
    i = low - 1
    for j in range(low, high):
        if arr[j] <= pivot:
            i += 1
            arr[i], arr[j] = arr[j], arr[i]
    arr[i + 1], arr[high] = arr[high], arr[i + 1]
    return i + 1

if __name__ == "__main__":
    main()

File: Practica6.java
import java.util.*;

public class Practica6 {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        int n = scanner.nextInt();
        scanner.nextLine();
        
        String[] medicinas = scanner.nextLine().split(" ");
        
        quicksort(medicinas, 0, n - 1);
        
        for (String medicina : medicinas) {
            System.out.print(medicina + " ");
        }
        
        scanner.close();
    }
    
    public static void quicksort(String[] arr, int low, int high) {
        if (low < high) {
            int pi = partition(arr, low, high);
            quicksort(arr, low, pi - 1);
            quicksort(arr, pi + 1, high);
        }
    }
    
    public static int partition(String[] arr, int low, int high) {
        String pivot = arr[high];
        int i = low - 1;
        for (int j = low; j < high; j++) {
            if (arr[j].compareTo(pivot) < 0) {
                i++;
                String temp = arr[i];
                arr[i] = arr[j];
                arr[j] = temp;
            }
        }
        String temp = arr[i + 1];
        arr[i + 1] = arr[high];
        arr[high] = temp;
        return i + 1;
    }
}